src/lib.rs

```diff
@@ -8,11 +8,14 @@ pub trait Timer<const TIMER_HZ: u32> {
     /// An error that might happen during waiting
     type Error: core::fmt::Debug;
 
+    /// The backing storage used for the timer. Can be either `u32` or `u64`.
+    type TimeStorage;
+
     /// Return current time `Instant`
-    fn now(&mut self) -> TimerInstantU32<TIMER_HZ>;
+    fn now(&mut self) -> Instant<Self::TimeStorage, 1, TIMER_HZ>;
 
     /// Start timer with a `duration`
-    fn start(&mut self, duration: TimerDurationU32<TIMER_HZ>) -> Result<(), Self::Error>;
+    fn start(&mut self, duration: Duration<Self::TimeStorage, 1, TIMER_HZ>) -> Result<(), Self::Error>;
```

Comment:
I don't think this actually makes sense, as this `Duration` will make it very hard for libraries to use, without specifying `T: Timer<TIMER_HZ, TimeStorage = u32>`, which kinda defeats the purpose.. 

My reasoning for wanting u64 backing storage, is that i have an 80 Mhz timer, and i still need to be able to support waits of both 50 microseconds and at least 3 minutes with the same timer. I think the better solution in my case it to change my Timer implementation to be 1 Mhz (divide the ticks by 80). That way i can have microsecond resolution, while still having up to roughly 71 minutes on u32, if my calculations are correct.

Reply:
If you need implementation example, you can see `counter.rs` at https://github.com/stm32-rs/stm32f4xx-hal/tree/master/src/fugit